sopel/tools/__init__.py

```diff
@@ -379,22 +420,29 @@ def check_pid(pid):
 
 
 def get_hostmask_regex(mask):
-    """Return a compiled `re.RegexObject` for an IRC hostmask"""
+    """Get a compiled regex pattern for an IRC hostmask
+
+    :param str mask: The hostmask that the pattern should match
+    :return: A compiled regex pattern matching the given ``mask``
+    :rtype: :py:class:`re.Pattern`
+    """
     mask = re.escape(mask)
     mask = mask.replace(r'\*', '.*')
     return re.compile(mask + '$', re.I)
 
 
 class SopelMemory(dict):
-
     """A simple thread-safe dict implementation.
 
-    *Availability: 4.0; available as ``Sopel.SopelMemory`` in 3.1.0 - 3.2.0*
-
     In order to prevent exceptions when iterating over the values and changing
     them at the same time from different threads, we use a blocking lock on
     ``__setitem__`` and ``contains``.
 
+    .. versionadded:: 3.1
```

Comment:
I think it's safe, by now, to remove this information. Since  6.x branch is the only one supported at the moment, anything older than 6.x doesn't need to be mentioned. Keep the `versionchanged` for 6.0, and remove what's from 3.1 and 4.0.

Reply:
Because old modules exist, I disagree on this one point. I made a point of finding out which version added these classes (both `SopelMemory` and `SopelMemoryWithDefault`) and adding the notations here to aid anyone who is trying to get a module written for an older version working again.

Actually, this is the only suggestion you made that I didn't take. :grin:

Reply:
My last argument [is this video](https://www.youtube.com/watch?v=yXyPvhISkRQ). But I'm pretty sure that won't convince you. 😆 

Don't worry, I think it's reasonable to keep that history.

Reply:
Nice clip!

I actually streamlined these in the second pass tonight, so they ought to look better in the docs and take up less space now.